e2e/nomostest/predicates.go

```diff
@@ -208,16 +209,16 @@ func HasCorrectResourceRequestsLimits(containerName string, cpuRequest, cpuLimit
 		}
 		for _, container := range dep.Spec.Template.Spec.Containers {
 			if containerName == container.Name {
-				if container.Resources.Requests[corev1.ResourceCPU] != cpuRequest {
+				if !equality.Semantic.DeepEqual(container.Resources.Requests[corev1.ResourceCPU], cpuRequest) {
```

Comment:
This looks like something that should be unit-testable. Can you add some unit tests to cover this bug fix?